blocks/inspector-controls/text-control/index.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from 'element';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+class TextControl extends Component {
+	constructor() {
+		super( ...arguments );
+		this.instanceId = this.constructor.instances++;
+	}
+
+	render() {
+		const { label, value, onChange } = this.props;
+		const id = 'inspector-text-control-' + this.instanceId;
+		const updateValue = ( event ) => onChange( event.target.value );
```

Comment:
I'm sympathetic to the [`jsx-no-bind` rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md)'s reasoning despite finding it to be far too restricting at times. At least when we already have a component class, I think it might be reasonable to prefer creating callbacks as bound instance members.